src/api/spec/models/package_spec.rb

```diff
@@ -264,4 +281,44 @@
       it{ expect(Package.valid_name?('_patchinfo')).to be(true) }
     end
   end
+
+  context "#buildresults" do
+    let(:results) { package.buildresults }
+    let(:results_test_package) { results['test_package'] }
+    let(:results_test_package_source) { results['test_package:test_package-source'] }
+
+    before do
+      allow(Buildresult).to receive(:find).and_return(fake_multibuild_results)
+    end
+
+    it { expect(results.keys).to match_array(['test_package', 'test_package:test_package-source']) }
+
+    it { expect(results_test_package.length).to eq(3) }
+
+    it { expect(results_test_package.first.repository).to eq('openSUSE_Leap_42.2') }
+    it { expect(results_test_package.first.architecture).to eq('x86_64') }
+    it { expect(results_test_package.first.code).to eq('succeded') }
+    it { expect(results_test_package.first.state).to eq('finished') }
+    it { expect(results_test_package.first.details).to be_nil }
+
+    it { expect(results_test_package.last.repository).to eq('openSUSE_Tumbleweed') }
+    it { expect(results_test_package.last.architecture).to eq('x86_64') }
+    it { expect(results_test_package.last.code).to eq('building') }
+    it { expect(results_test_package.last.state).to eq('building') }
+    it { expect(results_test_package.last.details).to be_nil }
+
+    it { expect(results_test_package_source.length).to eq(3) }
+
+    it { expect(results_test_package_source.first.repository).to eq('openSUSE_Leap_42.2') }
+    it { expect(results_test_package_source.first.architecture).to eq('x86_64') }
+    it { expect(results_test_package_source.first.code).to eq('disabled') }
+    it { expect(results_test_package_source.first.state).to eq('finished') }
+    it { expect(results_test_package_source.first.details).to be_nil }
+
+    it { expect(results_test_package_source.last.repository).to eq('openSUSE_Tumbleweed') }
+    it { expect(results_test_package_source.last.architecture).to eq('x86_64') }
+    it { expect(results_test_package_source.last.code).to eq('unresolvable') }
+    it { expect(results_test_package_source.last.state).to eq('building') }
+    it { expect(results_test_package_source.last.details).to be_nil }
```

Comment:
I guess it's okay if you only test the `.first` result. This repetition doesn't help in any way...